src/extension.ts

```diff
@@ -1941,12 +1941,30 @@ function isLintable(document: vscode.TextDocument): boolean {
     return document.languageId === 'yaml' || document.languageId === 'json' || document.languageId === 'helm';
 }
 
+function linterEnabled(enabledLinters: string[], name: string): boolean {
+    for (const linter of enabledLinters) {
+        if (linter === name) {
+            return true;
+        }
+    }
+    return false;
+}
+
 async function kubernetesLint(document: vscode.TextDocument): Promise<void> {
     // Is it a Kubernetes document?
     if (!isLintable(document)) {
         return;
     }
-    const linterPromises = linters.map((l) => l.lint(document));
+    const enabledLinters = config.getEnabledLinters();
+    if (enabledLinters.length === 0) {
+        return;
+    }
+    const linterPromises: Promise<vscode.Diagnostic[]>[] = [];
+    linters.map((l) => {
```

Comment:
Don't use `map` for side-effectful iteration.  `map` is for transforming one array to another.  I think you can write this as something like:

```
const linterPromises =
    linters.filter((l) => linterEnabled(enabledLinters, l.name()))
           .map((l) => l.lint(document));
```

(I would probably add an `isEnabled` helper function to make the `filter` more readable.)

Reply:
done.